arrow-cast/src/display.rs

```diff
@@ -866,6 +951,15 @@ pub fn lexical_to_string<N: lexical_core::ToLexical>(n: N) -> String {
 mod tests {
     use super::*;
 
+    const TEST_CONST_OPTIONS: FormatOptions<'static> = FormatOptions::new()
```

Comment:
I think a comment explaining the rationale of the test (so it isn't inadvertently remove) would be helpful

```suggestion
    // Test to verify options can be constant. See  
    // https://github.com/apache/arrow-rs/issues/4580
    const TEST_CONST_OPTIONS: FormatOptions<'static> = FormatOptions::new()
```